### DISCUSSION TITLE
`hc` $get will be found, but $post won't be found `404`

### DISCUSSION THREAD

**Discussion by roccho-dev** [honojs/hono#2147] [Q&A]
Forgive me if this question is too basic one and already solved
But I'm wandering this issue.

Why $post method with `hc` is not found of this project `./src/index.test.ts` despite that $get method will be success?
This is mostly same as [document](https://hono.dev/guides/rpc)

```
import { zValidator } from '@hono/zod-validator'
import { Hono } from 'hono'
import { hc } from 'hono/client'
import { describe, expect, test } from 'vitest'
import { z } from 'zod'

const app = new Hono()
const route = app
    .post('/posts', zValidator('form', z.object({ title: z.string(), body: z.string(), })), (c) => {
        return c.json({ ok: true, message: 'Created!', }, 201
        )
    })

    .get('/hello', (c) => c.text('hello'))

// fetching server is success
// this url is correct
const client = hc<typeof route>('http://127.0.0.1:5173')

describe('endpoint test', () => {

    // fail test
    test('/posts', async () => {
        const res = await client.posts.$post({
            form: {
                title: 'Hello',
                body: 'Hono is a cool project',
            },
        })
        console.debug('### res', await res)
        expect((await res).status).toBe(201)
    })

    // expected
    test('/hello', async () => {
        const res = await client.hello.$get()
        expect(await res.status).toBe(200)
    })

})
```

I've propared the env. [codesandbox](https://codesandbox.io/p/devbox/jolly-firefly-rz476m?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cls76j2b90009356hyj9syxij%2522%252C%2522sizes%2522%253A%255B36.37723689912423%252C63.62276310087577%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cls76j2b80002356hjrgq4oln%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cls76j2b80006356hc3i9v9cs%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cls76j2b80008356h8ujus9ym%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B100%252C0%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cls76j2b80002356hjrgq4oln%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls76j2b70001356h0sfk8i6n%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpnpm-lock.yaml%2522%257D%255D%252C%2522id%2522%253A%2522cls76j2b80002356hjrgq4oln%2522%252C%2522activeTabId%2522%253A%2522cls76j2b70001356h0sfk8i6n%2522%257D%252C%2522cls76j2b80008356h8ujus9ym%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls76j2b80007356hfk9ofw9s%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cls76j2b80008356h8ujus9ym%2522%252C%2522activeTabId%2522%253A%2522cls76j2b80007356hfk9ofw9s%2522%257D%252C%2522cls76j2b80006356hc3i9v9cs%2522%253A%257B%2522id%2522%253A%2522cls76j2b80006356hc3i9v9cs%2522%252C%2522activeTabId%2522%253A%2522cls76jbc3004f356hcyf4mghr%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls76j2b80003356hxgmykais%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522cls76j2b80004356ht8f9n4d4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cls75p8dv0007dtf9c0rfbqw1%2522%257D%252C%257B%2522id%2522%253A%2522cls76j2b80005356hva4o1q4i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cls76inup000edtf91i4n5vu5%2522%257D%252C%257B%2522id%2522%253A%2522cls76jbc3004f356hcyf4mghr%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cls76jb50000ldtf996w36fux%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Afalse%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

![image](https://github.com/honojs/hono/assets/40359643/0e87f7bd-30b0-4298-8816-3d32f1c604a6)

```
pnpm vitest ./src
```

**Comment by janat08** [comment]
I had a problem like this. What happened is that I had a route with params that was declared sooner than the post route. It's be cool if Hono tried to match against parameterized routes last.
